Formula/petsc.rb

```diff
@@ -0,0 +1,197 @@
+class Petsc < Formula
+  desc "Scalable solution of models that use partial differential equations"
+  homepage "https://www.mcs.anl.gov/petsc/index.html"
+  url "http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-lite-3.7.6.tar.gz"
+  sha256 "b07f7b4e57d75f982787bd8169f7b8debd5aee2477293da230ab6c80a52c6ef8"
+  revision 4
+  head "https://bitbucket.org/petsc/petsc", :using => :git
+
+  bottle :disable, "needs to be rebuilt with latest open-mpi"
+
+  option "without-test", "Skip build-time tests (not recommended)"
+  option "with-complex", "Link complex version of PETSc by default."
+  option "with-debug", "Build debug version"
+  option "with-ml", "Download and build ML (will not symlink if Trilinos is installed)"
+
+  depends_on "cmake" => :build
+  depends_on "gcc"
+  depends_on "open-mpi"
+  depends_on :x11 => :optional
+
+  depends_on "superlu"      => :recommended
+  depends_on "metis"        => :recommended
+  depends_on "scalapack"    => :recommended
+  depends_on "hypre"        => :recommended
+  depends_on "hdf5"         => ["with-mpi", :recommended]
+  depends_on "hwloc"        => :recommended
+  depends_on "suite-sparse" => :recommended
+  depends_on "netcdf"       => :recommended
+  depends_on "fftw"         => ["with-mpi", :recommended]
+
+  def oprefix(f)
+    Formula[f].opt_prefix
+  end
+
+  def install
+    ENV.deparallelize
```

Comment:
We consider `ENV.deparallelize a bug that needs to be reported and fixed upstream with an issue link in the formula.

Reply:
FWIW, PETSc's has a non-parallel legacy makefile that invokes parallel make on a GNU makefile. You can invoke GNU `make -f gmakefile` directly with your preferred parallel make flags. (This might be less forward-compatible if PETSc purges the legacy makefiles.)

Reply:
Thanks for the suggestion, @jedbrown